### PR TITLE
Update zion testing library and other jest packages.

### PR DESCRIPTION
**Pull request by marybeshaw**
I know the app template tests pass because Travis creates a new app, then runs tests, and those tests pass.

I updated to zion-testing-library v5 in my app and then npm-linked to react-scripts to test the jest setup.

Also, I updated the pin a minor version. **This may need to be a major pin update** because the jest-dom version was updated 2 versions and react-scripts is used directly by apps for jest.

### CODE REVIEW

File: packages/react-scripts/scripts/utils/createJestConfig.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ module.exports = (resolve, rootDir, isEjecting) => {
       '<rootDir>/packages/**/__tests__/**/*.{js,jsx,ts,tsx}',
       '<rootDir>/packages/**/*.{spec,test}.{js,jsx,ts,tsx}',
     ],
-    testEnvironment: 'jest-environment-jsdom-fourteen',
+    testEnvironment: 'jest-environment-jsdom-sixteen',
```
Comment (marybeshaw) [inline packages/react-scripts/scripts/utils/createJestConfig.js:51]: This is why this PR may need to be a major pin update.

File: packages/react-scripts/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@fs/react-scripts",
-  "version": "4.4.0",
+  "version": "4.5.0",
```
Comment (joeycozza) [inline packages/react-scripts/package.json:3]:
```suggestion
  "version": "5.0.0",
```

**Comment by marybeshaw** [comment]
It looks like a major pin update isn't needed.  The changelog shows it's not very likely to break something.
https://github.com/jsdom/jsdom/blob/master/Changelog.md

File: packages/react-scripts/package.json (hunk 2 above)
Comment (marybeshaw) [reply packages/react-scripts/package.json]: Why no pin change?  Won't that not publish it?

**Pull request merged**